kali_cloud_init.tf

```
@@ -62,4 +62,61 @@ data "cloudinit_config" "kali_cloud_init_tasks" {
     filename     = "mount-efs-share.sh"
     merge_type   = "list(append)+dict(recurse_array)+str()"
   }
+
+  # Create a credentials file for the VNC user that can be used to
+  # configure the AWS CLI to assume the findings data bucket write
+  # role.  For details, see
+  # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-a-configuration-file
+  #
+  # Input variables are:
+  # * aws_region - the AWS region where the roles are to be assumed
+  # * findings_data_bucket_write_role_arn - the ARN of the IAM role
+  #   that can be assumed to write to the findings data S3 bucket
+  # * permissions - the octal permissions to assign the AWS
+  #   configuration
+  # * vnc_read_parameter_store_role_arn - the ARN of the role that
+  #   grants read-only access to certain VNC-related SSM Parameter Store
+  #   parameters, including the VNC username
+  # * vnc_username_parameter_name - the name of the SSM Parameter Store
+  #   parameter containing the VNC user's username
+  part {
+    content = templatefile(
+      "${path.module}/cloud-init/write-kali-aws-config.tpl.sh", {
+        aws_region                          = var.aws_region
+        findings_data_bucket_write_role_arn = data.terraform_remote_state.sharedservices.outputs.assessment_findings_write_role.arn
+        permissions                         = "0400"
+        vnc_read_parameter_store_role_arn   = aws_iam_role.guacamole_parameterstorereadonly_role.arn
+        vnc_username_parameter_name         = var.ssm_key_vnc_username
```

This aligns with my above suggestions.

```suggestion
        vnc_username                        = data.aws_ssm_parameter.vnc_username.value
```

---

Please see commit 7896da0.